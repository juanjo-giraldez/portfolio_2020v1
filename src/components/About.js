import React from 'react';
import '../css/About.css';
import Background from '../img/Background.jpg';

export const About = () => {



    return (
        <div>
            <div>
                <h1 className="tamaño">Acerca de mí</h1>
            </div>
            <div className="info">
                <img className="info_image" src={Background} alt="imagen" />
                <div  className="contain_p">
                    <p>
                        Mi interés por aprender me llevo hasta al sector del Web Developing.
                        donde me inicie. Me propuse conocer las TICs , lanzandome a realizar
                        un Bootcamp en Desarrollo web y poseer experiencia en proyectos.
                       </p>
                    <p>Dispuesto a emprender nuevas aventuras por descubrir.....¿me
                    acompañas? .
                     </p>
                    <a
                        href="https://drive.google.com/file/d/1L2V9ZH55eLbw0I-cqoM5sPwS-Kci8ISi/view?usp=sharing"
                        target="_blank" rel="noopener noreferrer"
                    >resume</a>
                </div>
            </div>


        </div >

    )
}
