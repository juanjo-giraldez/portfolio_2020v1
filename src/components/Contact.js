import React from 'react';
import '../css/contact.css';

export const Contact = () => {
    return (
        <>


            <h1 className="title">Contacta conmigo</h1>

            <table className="contact">
                <tbody>
                    <tr>
                        <td>
                            <p>Email:</p>
                        </td>
                        <th><a href='mailto:jjgiraldezjurado@gmail.com'>jjgiraldezjurado@gmail.com</a></th>
                    </tr>
                    <tr>
                        <td>
                            <p>LinkedIn:</p>
                        </td>
                        <th><a href='https://www.linkedin.com/in/juanjogirÃ¡ldez'
                            target="_blank" rel="noopener noreferrer">juanjo-giraldez</a></th>
                    </tr>
                    <tr>
                        <td>
                            <p>Github:</p>
                        </td>
                        <th><a href='https://github.com/juanjo-giraldez'
                            target="_blank" rel="noopener noreferrer">juanjo-giraldez</a></th>
                    </tr>
                    <tr>
                        <td>
                            <p>Gitlab:</p>
                        </td>
                        <th><a href='https://gitlab.com/juanjogiraldez_kairosds'
                            target="_blank" rel="noopener noreferrer">juanjogiraldez_kairosds</a></th>
                    </tr>
                </tbody>
            </table>

        </>
    )
}


