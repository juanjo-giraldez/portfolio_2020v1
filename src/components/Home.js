import React from 'react';
import '../css/home.css';

export const Home = () => {
    return (
        <div className="hero">
            <div className="contenedor">
                <div className="contenido">
                    <h2 className="subtitulo">Hola,  Soy </h2>
                    <h1 className="titulo">Juanjo Gir√°ldez
                    </h1>

                </div>
            </div>

        </div>
    )
}
