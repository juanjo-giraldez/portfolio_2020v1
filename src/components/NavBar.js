import React from 'react';
import { Link, NavLink } from "react-router-dom";
import { useStateBar } from '../hooks/useStateBar';


export const NavBar = () => {

    const [state, open, reset, close] = useStateBar(false)

    return (
        <nav >
            <div >
                <Link to="/" onClick={reset}><svg className="logo" alt="logo"></svg></Link>
            </div>
            <div className="link">
                <div className={state ? "toggle" : "mobile"}>
                    <NavLink className="item" onClick={reset} exact to="/" >Home</NavLink>
                    <NavLink className="item" onClick={reset} exact to="./about" >Sobre m√≠</NavLink>
                    <NavLink className="item" onClick={reset} exact to="./contact" >Contacto</NavLink>
                    <NavLink className="item" onClick={reset} exact to="./skills" >Habilidades</NavLink>
                    <NavLink className="item" onClick={reset} exact to="./proyects"  >Proyectos</NavLink>
                    <div  onClick={close}  ><svg className=" close " alt="close"></svg></div>
                </div>
                <div onClick={open}><svg className="menu" alt="menu"></svg></div>
            </div>
        </nav>
    )
}

