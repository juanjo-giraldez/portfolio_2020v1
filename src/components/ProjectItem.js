
import React from 'react'

export const ProjectItem = ({ demo, id, img, project, technologies, url }) => {

    const result = technologies.map((e) => <h4 key={e}>{e}</h4>)
    
    return (
        <>
            <div className="container">
                <div className="card">
                    <div className="card_front">
                        <img src={img} alt={project} />
                        <h4> {project} </h4>
                    </div>
                    <div className="card_back">
                        <div className="place">
                            <h3>Tecnolog√≠as:</h3>
                            <div className="tech">{result}</div>
                        </div>
                        <div className="card_info">
                            <a href={url} target="_blank" rel="noopener noreferrer" >Repo</a>
                            {demo && <a href={demo} target="_blank" rel="noopener noreferrer">Demo</a>}
                        </div>
                    </div>
                </div>
            </div>
        </>
    )
}
