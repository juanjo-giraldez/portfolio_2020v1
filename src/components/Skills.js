import React from 'react';

export const Skills = () => {
    return (
        <div>
            <section className="contain">
                <div className="skill_tech">
                    <div>
                        <h3>FrontEnd</h3>
                        <ul>
                            <li className="html">HTML</li>
                            <li className="css3">CSS3</li>
                            <li className="sass">SASS</li>
                            <li className="react">REACT</li>
                            <li className="angular">ANGULAR</li>
                            <li className="vue">VUE</li>
                            <li className="bootstrap">BOOTSTRAP</li>
                            <li className="prime_vue">PRIME-VUE</li>
                            <li className="prime_ng">PRIME-NG</li>
                        </ul>
                    </div>

                    <div>
                        <h3>BackEnd</h3>
                        <ul>
                            <li className="javascript">JAVASCRIPT</li>
                            <li className="mongo_db">MONGO-DB</li>
                            <li className="mongoose">MONGOOSE</li>
                            <li className="apis">APIs</li>
                            <li className="node">NODE.js</li>
                            <li className="express">EXPRESS.js</li>
                            <li className="npm">NPM</li>
                            <li className="git">GIT</li>
                            <li className="docker">DOCKER</li>
                        </ul>
                    </div>
                </div>
            </section>
        </div>
    )
}
