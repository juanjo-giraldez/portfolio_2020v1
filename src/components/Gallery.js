import React from 'react';
import { useProjects } from '../hooks/useProjects';
import { ProjectItem } from './ProjectItem';

export const Gallery = () => {

    const { data } = useProjects()

    return (
        <div>
            <h1>Estos   son mis proyectos</h1>
            <div className="card-grid">
                {data.map((e) =>
                    (< ProjectItem key={e.id} {...e} />))}
            </div>
        </div>
    )
}