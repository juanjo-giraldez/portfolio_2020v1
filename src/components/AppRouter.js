import React from 'react';
import { BrowserRouter as Router, Redirect, Route, Switch } from "react-router-dom";
import { About } from "./About";
import { Gallery } from "./Gallery";
import { Home } from "./Home";
import { NavBar } from "./NavBar";

export const AppRouter = () => {
    return (
        <Router >
        <div>
            <NavBar />
            <div>

                <Switch>
                    <Route exact path="/" component={ Home } />
                    <Route exact path="/about" component={ About } />
                    <Route exact path="/proyects" component={ Gallery } />
                    <Redirect to="/" />
                </Switch>

            </div>
        </div>

    </Router>
    )
}
