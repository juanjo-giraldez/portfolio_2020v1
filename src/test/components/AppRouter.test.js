import { shallow } from 'enzyme';
import React from 'react';
import { AppRouter } from "../../components/AppRouter";

describe('Test in <About />', () => {

    it('should show success Web Component, <AppRouter />', () => {

        const wrapper =  shallow(<AppRouter />);

        expect(wrapper).toMatchSnapshot();
        
    })
    
    
})