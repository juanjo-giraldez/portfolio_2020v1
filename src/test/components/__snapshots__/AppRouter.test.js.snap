// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <About /> should show success Web Component, <AppRouter /> 1`] = `
<BrowserRouter>
  <NavBar />
  <Switch>
    <Route
      component={[Function]}
      exact={true}
      path="/"
    />
    <Route
      component={[Function]}
      exact={true}
      path="/about"
    />
    <Route
      component={[Function]}
      exact={true}
      path="/contact"
    />
    <Route
      component={[Function]}
      exact={true}
      path="/skills"
    />
    <Route
      component={[Function]}
      exact={true}
      path="/proyects"
    />
    <Redirect
      to="/"
    />
  </Switch>
</BrowserRouter>
`;
